transaction/transaction_test.go

```diff
@@ -0,0 +1,12 @@
+package transaction
+
+import "testing"
+
+func TestSettingFailToString(t *testing.T) {
+	trans := Transaction{}
+	trans.Fail = "Fail"
```

Comment:
Yea the reason for it was that since `Fail` was of interface type I was pretty sure it would work ok but wanted to be 100% sure.  I can remove it if you want.  It was mostly me just kind of exploring the interface type.